File: mobile/lib/models/notification.dart
class AppNotification {
  final String id;
  final String userId;
  final String title;
  final String body;
  final String type;
  final String? referenceId;
  final bool read;
  final DateTime createdAt;

  AppNotification({
    required this.id,
    required this.userId,
    required this.title,
    required this.body,
    required this.type,
    this.referenceId,
    required this.read,
    required this.createdAt,
  });

  factory AppNotification.fromJson(Map<String, dynamic> json) {
    return AppNotification(
      id: json['id'],
      userId: json['userId'],
      title: json['title'],
      body: json['body'],
      type: json['type'],
      referenceId: json['referenceId'],
      read: json['read'] ?? false,
      createdAt: DateTime.parse(json['createdAt']),
    );
  }
}

File: backend/src/main/java/com/assistbridge/model/Notification.java
package com.assistbridge.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "notifications")
public class Notification {
    @Id
    private String id;
    
    private String userId;
    
    private String title;
    
    private String body;
    
    private NotificationType type;
    
    private String referenceId;
    
    private boolean read;
    
    private LocalDateTime createdAt;
    
    public enum NotificationType {
        REQUEST_CREATED,
        REQUEST_ASSIGNED,
        VOLUNTEER_ACCEPTED,
        REQUEST_IN_PROGRESS,
        REQUEST_COMPLETED,
        REQUEST_CANCELLED
    }
}
